R/vol-surface-class.R

```diff
@@ -0,0 +1,160 @@
+#' VolSurface class
+#'
+#' The `VolSurface`` class is designed to represent volatility surfaces.
+#' Checks whether the dimentions of the surface correspond to the right type of axis: moneyness, delta or strike
+#'
+#' @param reference_date `Date` value that represents the as of date
+#' @param surface_type type of the vol dimentions. Available types: "strike/term"
+#' @param vol_quotes `matrix` of class `numeric` holding the vol values.
```

Comment:
we shouldnt be using matrices as an input type - they are terrible to work with. Use a (tidy) tibble. Suggest the tibble structure is something like:

```csv
maturity, strike, value
20200201, 72.00, 0.20
20200301, 72.00, 0.24
..., ..., ...
20200201, 82.00, 0.25
20200301, 82.00, 0.26
..., ..., ...
```

In the doc be very clear about the structure of this including for example that:

1. The name of the second column (e.g. `strike` or `moneyness` or `delta`) will define what sort of surface you are getting. Be clear about the options that are valid.
2. The string values / formats required for each field: e.g. `maturity` must be a string of format `YYYYMMDD`, `value` much be in decimals...etc.

Reply:
I changed the structure of VolSurface following the above suggestion. Now it only includes: reference_date, vol_quotes(tibble as described in the example above), ticker and Interpolation.